guacapy/client.py

```diff
@@ -483,3 +483,47 @@ def add_sharing_profile(self, payload, datasource=None):
             url="{}/session/data/{}/sharingProfiles".format(self.REST_API, datasource),
             payload=payload,
         )
+
+    def get_user_groups(self, datasource=None):
+        """
+        List User Groups
+        """
+        if not datasource:
+            datasource = self.primary_datasource
+        return self.__auth_request(
+            method="GET",
+            url="{}/session/data/{}/userGroups".format(self.REST_API, datasource,),
+        )
+
+    def get_group(self, usergroup, datasource=None):
+        """
+        Details of User Group
+        """
+        if not datasource:
+            datasource = self.primary_datasource
+        return self.__auth_request(
+            method="GET",
+            url="{}/session/data/{}/userGroups/{}".format(
+                self.REST_API, datasource, usergroup
+            ),
+        )
+
+    def add_member_user_group(self, usergroup, payload, datasource=None):
```

Comment:
`edit_group_members` may make more sense here since this method can be used to add or delete group members.